webpack/scenes/ContentViews/Publish/PublishContentViewWizard.js

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { STATUS } from 'foremanReact/constants';
+import { Wizard } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import CVPublishForm from './CVPublishForm';
+import CVPublishFinish from './CVPublishFinish';
+import getEnvironmentPaths from '../components/EnvironmentPaths/EnvironmentPathActions';
+import CVPublishReview from './CVPublishReview';
+import {
+  selectEnvironmentPaths,
+  selectEnvironmentPathsStatus,
+} from '../components/EnvironmentPaths/EnvironmentPathSelectors';
+
+const PublishContentViewWizard = ({
+  details, show, setIsOpen, currentStep, setCurrentStep,
+}) => {
+  const { name, id: cvId, version_count: versionCount } = details;
+  const [description, setDescription] = useState('');
+  // Change to userCheckedItems
+  const [userCheckedItems, setUserCheckedItems] = useState([]);
+  const [promote, setPromote] = useState(false);
+  const [forcePromote, setForcePromote] = useState([]);
+  const [envPathFlat, setEnvPathFlat] = useState([]);
+  const dispatch = useDispatch();
+  const environmentPathResponse = useSelector(selectEnvironmentPaths);
+  const environmentPathStatus = useSelector(selectEnvironmentPathsStatus);
+  const environmentPathLoading = environmentPathStatus === STATUS.PENDING;
+
+  const steps = [
+    {
+      id: 1,
+      name: 'Publish',
+      component: <CVPublishForm
+        description={description}
+        setDescription={setDescription}
+        details={details}
+        userCheckedItems={userCheckedItems}
+        setUserCheckedItems={setUserCheckedItems}
+        promote={promote}
+        setPromote={setPromote}
+        forcePromote={forcePromote}
+      />,
+    },
+    {
+      id: 2, name: 'Review', component: <CVPublishReview details={details} userCheckedItems={userCheckedItems} />, nextButtonText: 'Finish',
+    },
+    {
+      id: 3,
+      name: 'Finish',
+      component: <CVPublishFinish
+        description={description}
+        setDescription={setDescription}
+        userCheckedItems={userCheckedItems}
+        setUserCheckedItems={setUserCheckedItems}
+        forcePromote={forcePromote}
+        cvId={cvId}
+        versionCount={versionCount}
+        show={show}
+        setIsOpen={setIsOpen}
+        setCurrentStep={setCurrentStep}
+      />,
+      isFinishedStep: true,
+    },
+  ];
+
+  useEffect(
+    () => {
+      dispatch(getEnvironmentPaths());
+      setCurrentStep(1);
+    },
+    [],
+  );
+
+  useEffect(() => {
+    if (!environmentPathLoading) {
+      const { results } = environmentPathResponse || {};
+      setEnvPathFlat(results.map(result => result.environments).flatten());
```

Comment:
Why keep `envPathFlat` in state?  I feel like it would work fine as a regular function..

Reply:
It felt like an operation needed just once..

Reply:
Sounds like a job for [`useMemo`](https://reactjs.org/docs/hooks-reference.html#usememo), not `useState`!

Reply:
Updated to use useMemo..My first use of it :)